### DISCUSSION TITLE
No sound at startup

### DISCUSSION THREAD

**Discussion by Jean-Gabriel1982** [GrandOrgue/grandorgue#1323] [Help]
Hello all,

I've moved my organ in another location, and installed anything. 

But starting my computer and GO, I don't have any sound. 

To recover sound, I've tested two options : 
- plug and unplug the audio jack prose on the computer

- go to the audio settings of the computer ("Line out" is selected) and just click on "Digital output" then click again on "Line out"
![image](https://user-images.githubusercontent.com/20370038/209466828-caa76d5d-c6f0-4dd2-985a-a0a99bde4088.jpeg)

I don't understand...

**Comment by oleg68** [comment]
Hello. Merry Christmas!

1. What operation system are you using?
2. After you exit from GO, do you have sound in the system or in another application?

**Comment by Jean-Gabriel1982** [comment]
Hello Oleg, Merry Christmas !

I'm using Linux Mint Cinnamon Edition 21. 
I haven't tested if I have sound in another application...

Edit : after test I don't have any sound in another app

**Comment by oleg68** [comment]
Please 

1. Restart you mint computer and 
2. **without starting GO** go to the System settings -> Materiel -> Son
3. Select the `Line Out...`
4. Click on the `Test du son` button and click on each speaker. Could you hear anything?
5. If no, try to change the `Profil de sortie` and repeat the test

**Reply by Jean-Gabriel1982** [reply]
Hello !

Sorry for the late answer and Happy new year !!

I've tested what you've asked. 

When I start my computer without starting GO at startup, when I go to my sound settings, and test sound, there is sound. 

When I start GO after that, I have sound. 

But when GO automatically starts at computer startup, no sound, I have to make the manipulation I gave at my first message. 

![image](https://user-images.githubusercontent.com/20370038/210177475-34cd96d6-2d53-4b2e-ab99-a4c28c842c4f.jpeg)

![image](https://user-images.githubusercontent.com/20370038/210177543-02737637-6142-4428-8b11-d59e9494a72c.jpeg)

**Reply by Jean-Gabriel1982** [reply]
EDIT : When in settings, I select PortAudio and "Valider", GO freezes, I must "force to close"

EDIT 2 no, now I can select PA and GO does not freeze ... 
sometimes it do it ..

But even if PA is selected, I have no sound

**Reply by oleg68** [reply]
``RtAudio: alsa: default`` means audio output through the `PulseAudio` server that you has suspended with pasuspender, so it can not produce any sound.

What devices are listed in the `Change` dialog?

**Reply by Jean-Gabriel1982** [reply]
Here is the list

![image](https://user-images.githubusercontent.com/20370038/210179593-a3a05711-29b1-4227-a187-d488017b0a19.jpeg)

And "Jack native output"

**Reply by oleg68** [reply]
Please try ``hw:HD-Audio Generic,1``. If it does not sound, try another `Generic`s.

(I think these generics correspond to different audio profiles: stereo, 5.1, 7.1 etc)

**Comment by midimusic** [comment]
Could it be that Jack is causing problems. I am not sure that it is all that compatible with Alsa especially if you do not fully understand what you are doing and have not set it up correctly. I suggest unticking all references to Jack in the GO setup and using the Default Alsa settings. In my experience those "Generic" settings are spurious and do not work.

I do not understand myself how GO audio can be OK if GO is started manually and not OK if GO is started auromatically when the computer starts. The difference must be in the order of events.

csw900

**Reply by rousseldenis** [reply]
I can confirm that GO does not answer correctly if you launch it directly at OS start.

As GO is reserving audio resource (as far as I've understood until now), and as that resource is busy during startup, GO cannot reserve it. I remember a discussion with @oleg68 recently on that point.

So, @Jean-Gabriel1982  I suggest adding a delay before starting GO at startup.

**Reply by oleg68** [reply]
> Could it be that Jack is causing problems.

Jack could not cause any problems until the jack server is started or the `Jack native` device is selected. But I agree with your suggestion to uncheck all audio interface that are not used.

**Reply by oleg68** [reply]
The root cause of the issue is quite clear: `alsa: default` works through pulseaudio, but @Jean-Gabriel1982 had suspended it before starting GO. So `alsa: default` doesn't work and another alsa device has to be used instead.

**Reply by Jean-Gabriel1982** [reply]
> The root cause of the issue is quite clear: `alsa: default` works through pulseaudio, but @Jean-Gabriel1982 had suspended it before starting GO. So `alsa: default` doesn't work and another alsa device has to be used instead.

Hello, maybe I have to delete the suspension of PA. 
Today I've tested all other lines and nothing works ... or I have to restart the computer after saving changes ?

Or jack causes trouble as it was said before ..

**Comment by oleg68** [comment]
@Jean-Gabriel1982 

> Hello, maybe I have to delete the suspension of PA.

If you don't have any succes, then remove pasuspender and stop pulseaudio manually:

```
systemctl stop --user pulseaudio.socket 
systemctl stop --user pipewire.socket 
```

Then start GO, deselect PortAudio and Jack, exit from GO and start it again. Then try to use all devices, available for RtAudio::Alsa.

**Reply by Jean-Gabriel1982** [reply]
Ok I'll do that. 
The problem is when GO is started I don't have sound. But if I close it and reopen, sound is here ... with RTAlsa Default selected ..

**Reply by Jean-Gabriel1982** [reply]
I removed "pasuspender" from the text file for GO starts at computer startup and it seems sound works fine now

![image](https://user-images.githubusercontent.com/20370038/211207649-4c8b19e9-fdca-4db2-842f-db62978fb819.jpeg)

![image](https://user-images.githubusercontent.com/20370038/211207687-6fe20987-0b32-4312-b730-18ff78d9315d.jpeg)

**Reply by oleg68** [reply]
> I removed "pasuspender" from the text file for GO starts at computer startup and it seems sound works fine now

Yes, but the latency is big. So it is better to remove pulseaudio.

**Reply by Jean-Gabriel1982** [reply]
> > I removed "pasuspender" from the text file for GO starts at computer startup and it seems sound works fine now
> 
> 
> 
> Yes, but the latency is big. So it is better to remove pulseaudio.

What do you mean by the latency ?

**Reply by oleg68** [reply]
Latency - the time between pressing a key and appearency of the sound.

Pulseaudio 1. adds an additional latency. 2. does not allow to decrease the buffer as little asalsa does.

**Reply by Jean-Gabriel1982** [reply]
> Latency - the time between pressing a key and appearency of the sound.
> 
> Pulseaudio 1. adds an additional latency. 2. does not allow to decrease the buffer as little asalsa does.

Ah ok. 
I remember some weeks/year ago
we add pasuspender at the startup file as GO worked weird with PulseAudio. 
We configure with RT which worked fine. 

Everything worked fine before I moved the organ in my church. I don't understand why it does not work now ..

I will remove PulseAudio ...

EDIT : I put "9000" in the icon keyboard on the main bar of GO. Before it was "5800". 
I think I can put again "5800". I tested if it changes something but it seems not. 

EDIT 2 : I have to tell that when GO starts, if I go in the settings and select or deselect some Audio Port, GO freezes and a popup appears asking me to wait the program respond or force to quit. 
If I force to quit and reopen GO I can select or deselect Audio Port without crash ..
